app/controllers/api/v1/lobbies_controller.rb

```diff
@@ -75,6 +94,34 @@ def require_authorisation
 
       def set_quiz
         @quiz = Quiz.find(params[:quiz_id])
+
+        return head :not_found if @quiz.nil?
+      end
+
+      def set_player
+        @player = Player.find(params[:player_id])
+
+        return head :not_found if @player.nil?
+      end
+
+      def set_lobby
+        @lobby = Lobby.find(params[:id])
+
+        return head :not_found if @lobby.nil?
+      end
+
+      def set_current_question
+        @question = @lobby.quiz.questions.detect { |question| question.order == @lobby.current_question_index }
+
+        return head :not_found if @question.nil?
+      end
+
+      def replace_player_answers
+        @question.player_answers.where(player_id: @player.id).delete_all
```

Comment:
Use `destroy_all` instead of `delete_all` as it will also run any callbacks that are on the model. `delete_all` simply executes a `DELETE` statement without taking into account anything else defined on the model.